src/Symfony/Component/Security/Core/Event/VoteEvent.php

```diff
@@ -28,11 +29,20 @@ final class VoteEvent extends Event
     private $attributes;
     private $vote;
 
-    public function __construct(VoterInterface $voter, $subject, array $attributes, int $vote)
+    /**
+     * @param Vote|int $vote
+     */
+    public function __construct(VoterInterface $voter, $subject, array $attributes, $vote)
```

Comment:
is it allowed to do so regarding BC and code evolution?
what about creating a new event class, and mark this one as deprecated in 5.4?

Reply:
I honestly don't know. The code is copied "as is" from the other PR (since the original contributor is busy now). I'm willing to make any changes to comply with the standards. Please let me know your recommendation.

Reply:
yes I know you are continuing the work and thank you for that :)
perhaps for this point we should wait a core member to give us the `reason` 🥁

Reply:
@stof Can you help to give some recommendations on how to handle this? Thank you very much.

Reply:
BC-wise, this specific change is fine: it's a final constructor so there is no potential signature mismatch, and it keeps handling both types.